### PR TITLE
Docs for SVM in new interfaces

### PR DESCRIPTION
**Pull request by PetrovKP**
# Description
Please include a summary of the change. For large or complex changes please include enough information to introduce your change and explain motivation for it.

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: cpp/oneapi/dal/algo/svm/common.hpp [hunk 1]
```diff
@@ -83,11 +97,33 @@ class descriptor_base : public base {
...
+    /// @invariant :expr:`cache_size >= 0.0`
+    /// @remark default = 200.0
     double get_cache_size() const;
+
+    /// The parameter of the WSS scheme $\\tau$.
+    /// @invariant :expr:`tau >= 0.0`
+    /// @remark default = 1e-6
     double get_tau() const;
+
+    /// A flag that enables use of a shrinking optimization technique. Used with :expr:`method::thunder` split finding method only.
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/common.hpp:125]:
```suggestion
    /// A flag that enables the use of a shrinking optimization technique. Used with :expr:`method::thunder` split-finding method only.
```

File: cpp/oneapi/dal/algo/svm/common.hpp [hunk 2]
```diff
@@ -196,38 +246,53 @@ class model : public base {
...
+    /// The bias
+    /// @remark default = 0.0
     double get_bias() const;
 
     auto& set_bias(double value) {
         set_bias_impl(value);
         return *this;
     }
 
+    /// The first unique value of class from labels
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/common.hpp:285]: maybe like this?
```suggestion
    /// The first unique value in class labels
```

File: cpp/oneapi/dal/algo/svm/common.hpp [hunk 3]
```diff
@@ -196,38 +246,53 @@ class model : public base {
...
+    /// @remark default = 0
     std::int64_t get_first_class_label() const;
 
     auto& set_first_class_label(std::int64_t value) {
         set_first_class_label_impl(value);
         return *this;
     }
 
+    /// The second unique value of class from labels
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/common.hpp:294]: maybe like this?
```suggestion
    /// The second unique value in class labels
```

File: cpp/oneapi/dal/algo/svm/infer_types.hpp [hunk 4]
```diff
@@ -36,22 +36,30 @@ using v1::infer_result_impl;
...
     static_assert(detail::is_valid_task_v<Task>);
 
 public:
     using task_t = Task;
 
+    /// Creates a new instance of the class with the given :literal:`model`
+    /// and :literal:`data` property values
     infer_input(const model<Task>& trained_model, const table& data);
 
+    /// The trained Decision Forest model
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/infer_types.hpp:52]:
```suggestion
    /// The trained SVM model
```

File: cpp/oneapi/dal/algo/svm/train_types.hpp [hunk 5]
```diff
@@ -36,29 +36,39 @@ using v1::train_result_impl;
 
 namespace v1 {
 
+/// @tparam Task   Tag-type that specifies type of the problem to solve. Can
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/train_types.hpp:39]:
```suggestion
/// @tparam Task   Tag-type that specifies the type of the problem to solve. Can
```

File: cpp/oneapi/dal/algo/svm/train_types.hpp [hunk 6]
```diff
@@ -36,29 +36,39 @@ using v1::train_result_impl;
...
+    /// The training set $X$
+    /// @remark default = table{}
     const table& get_data() const;
 
     auto& set_data(const table& value) {
         set_data_impl(value);
         return *this;
     }
 
+    /// Vector of labels $y$ for the training set $X$
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/train_types.hpp:61]:
```suggestion
    /// The vector of labels $y$ for the training set $X$
```

File: cpp/oneapi/dal/algo/svm/train_types.hpp [hunk 7]
```diff
@@ -36,29 +36,39 @@ using v1::train_result_impl;
...
+    /// @remark default = table{}
     const table& get_labels() const;
 
     auto& set_labels(const table& value) {
         set_labels_impl(value);
         return *this;
     }
 
+    /// Vector of weights $w$ for the training set $X$
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/train_types.hpp:70]:
```suggestion
    /// The vector of weights $w$ for the training set $X$
```

File: cpp/oneapi/dal/algo/svm/train_types.hpp [hunk 8]
```diff
@@ -75,45 +85,61 @@ class train_input : public base {
     dal::detail::pimpl<detail::train_input_impl<Task>> impl_;
 };
 
+/// @tparam Task Tag-type that specifies type of the problem to solve. Can
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/train_types.hpp:88]:
```suggestion
/// @tparam Task Tag-type that specifies the type of the problem to solve. Can
```

File: cpp/oneapi/dal/algo/svm/train_types.hpp [hunk 9]
```diff
@@ -75,45 +85,61 @@ class train_input : public base {
...
+    /// @remark default = model<Task>{}
     const model<Task>& get_model() const;
 
     auto& set_model(const model<Task>& value) {
         set_model_impl(value);
         return *this;
     }
 
+    /// A $nsv \\times p$ table containing support vectors.
+    /// Where $nsv$ - number of support vectors.
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/train_types.hpp:114]:
```suggestion
    /// A $nsv \\times p$ table containing support vectors,
    /// where $nsv$ is the number of support vectors.
```

File: docs/source/onedal/svm/svm.rst [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+
+.. default-domain:: cpp
+
+.. _alg_svm:
+
+=======================================
+Support Vector Machine Classifier (SVM)
+=======================================
+
+Support Vector Machine (SVM) :capterm:`classification`  is among popular classification
```
Comment (outoftardis) [inline docs/source/onedal/svm/svm.rst:25]:
```suggestion
Support Vector Machine (SVM) :capterm:`classification` is among popular classification
```

File: docs/source/onedal/svm/svm.rst [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+algorithms. It belongs to a family of generalized linear classification problems.
+
+
+.. |t_math| replace::    `Training <svm_t_math_>`_
+.. |t_thunder| replace:: `Thunder <svm_t_math_thunder_>`_
+.. |t_smo| replace::     `Smo <svm_t_math_smo_>`_
```
Comment (outoftardis) [inline docs/source/onedal/svm/svm.rst:31]:
```suggestion
.. |t_smo| replace::     `SMO <svm_t_math_smo_>`_
```

File: docs/source/onedal/svm/svm.rst [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+.. |t_input| replace::   `train_input <svm_t_api_input_>`_
+.. |t_result| replace::  `train_result <svm_t_api_result_>`_
+.. |t_op| replace::      `train(...) <svm_t_api_>`_
+
+.. |i_math| replace::    `Inference <svm_i_math_>`_
+.. |i_smo| replace::     `Smo <svm_i_math_smo_>`_
```
Comment (outoftardis) [inline docs/source/onedal/svm/svm.rst:37]:
```suggestion
.. |i_smo| replace::     `SMO <svm_i_math_smo_>`_
```

File: docs/source/onedal/svm/svm.rst [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+      -  **Cache**:      
+         the implementation can allocate a predefined amount of memory
+         to store intermediate results of the kernel computation.
+
+      -  **Shrinking**:
+         the implementation can try to decrease the amount of kernel
+         related computations (see [Joachims99]_).
+
+The solution of the problem defines the separating hyperplane and
+corresponding decision function :math:`D(x)= \sum_{k} {y_k \alpha_k K(x_k, x)} + b`
```
Comment (outoftardis) [inline docs/source/onedal/svm/svm.rst:89]:
```suggestion
corresponding decision function :math:`D(x)= \sum_{k} {y_k \alpha_k K(x_k, x)} + b`,
```

File: docs/source/onedal/svm/svm.rst [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+corresponding decision function :math:`D(x)= \sum_{k} {y_k \alpha_k K(x_k, x)} + b` 
+where only those :math:`x_k` that correspond to non-zero :math:`\alpha_k` appear in the sum,
+and :math:`b` is a bias. Each non-zero :math:`\alpha_k` is called a classification
+coefficient and the corresponding :math:`x_k` is called a support vector.
+
+.. _svm_t_math_smo:
+
+Training method: *smo*
+~~~~~~~~~~~~~~~~~~~~~~
+In *smo* training method, the all vectors from training dataset is used for each iteration.
```
Comment (outoftardis) [inline docs/source/onedal/svm/svm.rst:98]:
```suggestion
In *smo* training method, all vectors from the training dataset are used for each iteration.
```

File: docs/source/onedal/svm/svm.rst [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+applying Sequential Minimal Optimization (SMO) solver to the selected subproblem.
+The description of this method is given in Algorithm [Wen2018]_. 
+
+.. _svm_i_math:
+.. _svm_i_math_smo:
+.. _svm_i_math_thunder:
+
+Inference methods: *smo* and *thunder*
+--------------------------------------
+*smo* and *thunder* inference methods performs prediction by the same way:
```
Comment (outoftardis) [inline docs/source/onedal/svm/svm.rst:115]:
```suggestion
*smo* and *thunder* inference methods perform prediction in the same way:
```

File: docs/source/onedal/svm/svm.rst [hunk 16]
```diff
@@ -0,0 +1,188 @@
...
+
+Given the SVM classifier and :math:`r` feature vectors :math:`x_1, \ldots, x_r`, 
+the problem is to calculate the signed value of the
+decision function :math:`D(x_i)`, :math:`i=1, \ldots, r`. The sign of the
+value defines the class of the feature vector, and the absolute
+value of the function is a multiple of the distance between the
+feature vector and separating hyperplane.
```
Comment (outoftardis) [inline docs/source/onedal/svm/svm.rst:122]:
```suggestion
feature vector and the separating hyperplane.
```

File: docs/source/onedal/svm/svm.rst [hunk 17]
```diff
@@ -0,0 +1,188 @@
...
+
+---------------------
+Programming Interface
+---------------------
+All types and functions in this section shall be declared in the
```
Comment (outoftardis) [inline docs/source/onedal/svm/svm.rst:127]:
```suggestion
All types and functions in this section are declared in the
```

File: docs/source/onedal/svm/svm.rst [hunk 18]
```diff
@@ -0,0 +1,188 @@
...
+``oneapi::dal::svm`` namespace and be available via inclusion of the
```
Comment (outoftardis) [inline docs/source/onedal/svm/svm.rst:128]:
```suggestion
``oneapi::dal::svm`` namespace and are available via inclusion of the
```

File: docs/source/onedal/svm/svm.rst [hunk 19]
```diff
@@ -0,0 +1,188 @@
...
+
+.. _svm_i_api_result:
+
+Result
+~~~~~~
+.. onedal_class:: oneapi::dal::svm::v1::infer_result
+
+Operation
+~~~~~~~~~
+.. onedal_func:: oneapi::dal::v1::infer
```
Comment (outoftardis) [inline docs/source/onedal/svm/svm.rst:188]: Operations should be included like this (same for train):

image

File: docs/source/onedal/svm/svm.rst [hunk 20]
```diff
@@ -186,3 +196,11 @@ Result
 Operation
 ~~~~~~~~~
 .. onedal_func:: oneapi::dal::v1::infer
```
Comment (outoftardis) [inline docs/source/onedal/svm/svm.rst:198]: you forgot to delete this line

Comment (outoftardis) [reply docs/source/onedal/svm/svm.rst:198]:
```suggestion
```

**Review by outoftardis — APPROVED** [review]
I'm ok with merging, docbuild warnings will be fixed by #1326

**Comment by PetrovKP** [comment]
@outoftardis Sorry, I forgot to commit docs for kernel functions. I think in #1326 need add sections for kernel functions

**Comment by outoftardis** [comment]
> @outoftardis Sorry, I forgot to commit docs for kernel functions. I think in #1326 need add sections for kernel functions

ok, almost done

@PetrovKP  could you please change underscores `_` to dashes `-` in rst files and folders? thanks

File: docs/source/onedal/kernel-functions/linear-kernel.rst [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+any pair of input vectors: :math:`K(x_i, y_i) = k {X_i}^T y_i + b`.
+
+.. _linear_kernel_c_math_dense:
+
+---------------------
+Programming Interface
+---------------------
+All types and functions in this section shall be declared in the
+``oneapi::dal::linear_kernel`` namespace and be available via inclusion of the
+``oneapi/dal/algo/linear_kernel.hpp`` header file.
```
Comment (outoftardis) [inline docs/source/onedal/kernel-functions/linear-kernel.rst:61]:
```suggestion
All types and functions in this section are declared in the
``oneapi::dal::linear_kernel`` namespace and are available via inclusion of the
``oneapi/dal/algo/linear_kernel.hpp`` header file.
```

File: docs/source/onedal/kernel-functions/rbf-kernel.rst [hunk 22]
```diff
@@ -0,0 +1,106 @@
...
+
+
+.. _rbf_kernel_c_math_dense:
+
+---------------------
+Programming Interface
+---------------------
+All types and functions in this section shall be declared in the
+``oneapi::dal::rbf_kernel`` namespace and be available via inclusion of the
+``oneapi/dal/algo/rbf_kernel.hpp`` header file.
```
Comment (outoftardis) [inline docs/source/onedal/kernel-functions/rbf-kernel.rst:65]:
```suggestion
All types and functions in this section are declared in the
``oneapi::dal::rbf_kernel`` namespace and are available via inclusion of the
``oneapi/dal/algo/rbf_kernel.hpp`` header file.
```

**Comment by outoftardis** [comment]
no other comments from me

**Pull request merged**